File: VirtualWorld-Java/src/Owca.java
import java.util.ArrayList;

public class Owca extends Zwierze{

public Owca() {
}
	
	public Owca(Swiat swiat, ArrayList<Integer> poz, int wiek, int inicj, int moc) {
		changeSwiat(swiat);
		changePolozenie(poz.get(0), poz.get(1));
		changeWiek(wiek);
		changeInicjatywa(inicj);
		
		String s = ""+'O';
		getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
	}
	
	public Owca(Swiat swiat, ArrayList<Integer> poz) {
		changeSwiat(swiat);
		changePolozenie(poz.get(0), poz.get(1));
		changeInicjatywa(4);
		zmienSile(4);
		changeWiek(0);
		
		String s = ""+'O';
		getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
	}
	
	@Override
	public void rysowanie() {
		System.out.println('O');
	}
	
	@Override
	public char getZnak() {
		return 'O';
	}
	
}

File: VirtualWorld-Cpp/Virtual-World/Owca.cpp
#include "Owca.h"
#include "Zwierze.h"
#include <iostream>
#include "Organizm.h"

Owca::Owca()
{
}

Owca::Owca(Swiat* swiat, std::vector<unsigned int> poz, int wiek, int inicj, int moc)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeWiek(wiek);
	changeInicjatywa(inicj);

	getSwiat()->edytujPlansze(poz[0], poz[1], OWCA);
}

Owca::Owca(Swiat* swiat, std::vector<unsigned int> poz)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeInicjatywa(4);
	zmienSile(4);
	changeWiek(0);

	getSwiat()->edytujPlansze(poz[0], poz[1], OWCA);
}

void Owca::rysowanie()
{
	std::cout << OWCA;
}

char Owca::getZnak()
{
	return OWCA;
}

Owca::~Owca()
{
}